gitclone/checkout.go

```diff
@@ -269,12 +269,31 @@ func createCheckoutStrategy(checkoutMethod CheckoutMethod, cfg Config, patchFile
 
 }
 
-func selectFetchOptions(checkoutStrategy CheckoutMethod, cloneDepth int, fetchTags, fetchSubmodules bool) fetchOptions {
-	return fetchOptions{
+func selectFetchOptions(checkoutStrategy CheckoutMethod, cloneDepth int, fetchTags, fetchSubmodules bool, filterTree bool) fetchOptions {
+	opts := fetchOptions{
 		depth:           cloneDepth,
 		tags:            fetchTags,
 		fetchSubmodules: fetchSubmodules,
 	}
+
+	opts = selectFilterTreeFetchOption(checkoutStrategy, opts, filterTree)
+
+	return opts
+}
+
+func selectFilterTreeFetchOption(checkoutStrategy CheckoutMethod, opts fetchOptions, filterTree bool) fetchOptions {
+	if !filterTree {
+		return opts
+	}
+
+	switch checkoutStrategy {
+	case CheckoutCommitMethod, CheckoutTagMethod, CheckoutBranchMethod:
```

Comment:
Can filtertree be used in case of PRs?

Reply:
I'm not sure, will check it.

Reply:
I'm not sure if all the various PR strategies (`CheckoutPRMergeBranchMethod`, `CheckoutPRManualMergeMethod`, `CheckoutPRDiffFileMethod`) would work with this approach. My gut feeling is there could be scenarios where it could work but I'm pretty sure there would be cases where it won't. 

My suggestion would be to keep sparse-checkout for the simple cases where a commit, tag or branch should be checked out and built.

Reply:
CheckoutPRMergeBranchMethod, CheckoutPRManualMergeMethod both fetch a branch then merge an other branch, which is pretty common scenario and can be tested. I agree that CheckoutPRDiffFileMethod may not work, I am ok if we do not support it for now.